mqtt_as/mqtt_as.py

```diff
@@ -406,14 +429,20 @@ def close(self):
             if sz != b"\x02":
                 raise OSError(-1)
             rcv_pid = await self._as_read(2)
+            self.dprint("received PUBACK", rcv_pid)
             self.rcv_pids.add(rcv_pid[0] << 8 | rcv_pid[1])
+            self.dprint("rcv_pids:", self.rcv_pids)
             # Discard old pid's: can arise if a publish fails before PUBACK arrives
-            self.rcv_pids = {x for x in self.rcv_pids if  (x - self.pid) % 65536 < 50}
+            # self.rcv_pids = {x for x in self.rcv_pids if (x - self.pid) % 65536 < 50}
+            self.dprint("rcv_pids after pruning:", self.rcv_pids)
 
         if op == 0x90:  # SUBACK
+            self.dprint("received suback")
             resp = await self._as_read(4)
             if resp[1] != self.pkt[2] or resp[2] != self.pkt[3] or resp[3] == 0x80:
+                self.dprint("suback oserror")
                 raise OSError(-1)
+            self.dprint("suback true")
             self.suback = True
```

Comment:
Since all subscribe check for self.suback, they will all return "succesfully" once the first subscribe receives a SUBACK and self.suback is set to True.
See comment in subscribe()